ftl/common/builder.py

```diff
@@ -75,21 +75,19 @@ def CreatePackageBase(self, base_image, cache):
     def BuildAppLayer(self):
         """Override."""
         buf = cStringIO.StringIO()
-        logging.info('Starting to generate tarfile from context...')
+        logging.info('Starting to generate app layer tarfile from context...')
         with tarfile.open(fileobj=buf, mode='w') as out:
             for name in self._ctx.ListFiles():
                 content = self._ctx.GetFile(name)
                 info = tarfile.TarInfo(os.path.join('app', name))
                 info.size = len(content)
                 out.addfile(info, fileobj=cStringIO.StringIO(content))
-        logging.info('Finished generating tarfile from context.')
+        logging.info('Finished generating app layer tarfile from context.')
 
         tar = buf.getvalue()
         sha = 'sha256:' + hashlib.sha256(tar).hexdigest()
 
-        gz = cStringIO.StringIO()
-        logging.info('Starting to gzip tarfile...')
-        with gzip.GzipFile(fileobj=gz, mode='w', compresslevel=1) as f:
-            f.write(tar)
+        logging.info('Starting to gzip app layer tarfile...')
+        subprocess.check_call(['gzip', tar])
```

Comment:
'tar' is just a string at this point. You'll have to pass it into stdin of subprocess via subprocess.communicate:
https://stackoverflow.com/questions/8475290/how-do-i-write-to-a-python-subprocess-stdin